### PR TITLE
Expand abbreviated property name

### LINKED ISSUE #16721: Rename `matl_timestep_limit` to  `material_timestep_limit`
## Reason
Saving four characters here is not worth the loss of clarity and we have a related parameter `use_material_timestep_limit`, which spells it out.

## Design
Just rename the property.

## Impact
Clarity

### PR DESCRIPTION
**Pull request by dschwen**
This will require changes in at least Blackbear and Bison.

Closes #16721

### CODE REVIEW

**Comment by dschwen** [comment]
Any @hugary1995  ...

**Comment by hugary1995** [comment]
Well, am I involved in this? Anyways, at a first glance, such a one-liner change requiring 10 files to be changed (I know some are documentation) makes me think if we can provide a MaterialTimeStepLimitInterface.

**Comment by dschwen** [comment]
@hugary1995 having merged your PR created a merge conflict for my PR. That's the extent of your involvement. I'm not against the idea of an interface here, but I think that'd be overkill. If we decided to rename the interface we'd have just as many edits :-)

File: modules/tensor_mechanics/src/materials/ComputeDamageStress.C [hunk 1]
```diff
@@ -24,7 +24,7 @@ ComputeDamageStress::validParams()
 
 ComputeDamageStress::ComputeDamageStress(const InputParameters & parameters)
   : ComputeFiniteStrainElasticStress(parameters),
-    _matl_timestep_limit(declareProperty<Real>("matl_timestep_limit")),
+    _material_timestep_limit(declareProperty<Real>("material_timestep_limit")),
```
Comment (tophmatthews) [inline modules/tensor_mechanics/src/materials/ComputeDamageStress.C:27]: No deprecation? Bold. I always hated "matl"

**Review by loganharbour — APPROVED** [review]

File: modules/tensor_mechanics/src/materials/ComputeDamageStress.C (hunk 1 above)
Comment (dschwen) [reply modules/tensor_mechanics/src/materials/ComputeDamageStress.C:27]: tests pass, so ...

**Comment by dschwen** [comment]
I have done that and have a PR ready. Issue is open. I didn't get around to
submitting it yet.

**Pull request merged**